independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/AnnotationsTransformationContext.java

```diff
@@ -27,9 +29,13 @@ abstract class AnnotationsTransformationContext<C extends Collection<AnnotationI
      * @param annotations Mutable collection of annotations
      */
     public AnnotationsTransformationContext(BuildContext buildContext, AnnotationTarget target,
+            AnnotationTarget methodParameterTarget,
             C annotations) {
         this.buildContext = buildContext;
         this.target = target;
+        // once we remove #getTarget(), 'target' field should contain method parameter AnnotationTarget for method parameter injection
```

Comment:
`getTarget()` is not deprecated, it probably should be? (Together with `getAllAnnotations()` maybe?)

Reply:
`getTarget` is an implementation of the already deprecated `InjectionPointsTransformer$TransformationContext#getTarget` and same goes for the `getAllAnnotations` method.

This is just a common abstract class we use for both, injection point transformers as well as observer method variant. However, the OM variant doesn't have the `getTarget` method at all and on top of that it has the `getAllAnnotations` which needs to stay non-deprecated as that's still relevant (you want annotations for whole OM as that's the annotation target in that case).

Does that make sense?

Reply:
Ouch! Yeah, makes sense, sorry for the noise.